tests/boot/boot_to_desktop.pm

```diff
@@ -42,4 +42,30 @@ sub test_flags {
     return {fatal => 1, milestone => 1};
 }
 
+sub post_fail_hook {
+    my $self = shift;
+
+    save_screenshot;
+
+    # check for text login to check if X has failed
+    if (check_screen('generic-login')) {
+        record_info 'Seems that the display manager failed';
+    }
+
+    # if we found a shell, we do not need the memory dump
+    if (!(match_has_tag('emergency-shell') or match_has_tag('emergency-mode'))) {
+        die "save_memory_dump is temporarily unavailable, see https://progress.opensuse.org/issues/19390";
```

Comment:
As this `die` is executed without any condition, the rest of the commands in the `if` block won't ever get executed.

Reply:
You're right, but this code is from first_boot so we should also change it there

Reply:
Then we should maybe reuse the code.